docs/source/migration.mdx

```diff
@@ -620,36 +626,61 @@ class MyDataSource extends RESTDataSource {
+In Apollo Server 2, certain poorly formatted requests receive HTTP responses with a 5xx status (indicating a server error) instead of 4xx (indicating a client error). For example, this occurs in some integrations for a missing POST body or a JSON parse error.
+
+In Apollo Server 3, these errors are handled in a more consistent manner across integrations and consistently return HTTP responses with a 4xx status.
 
 ### Extensions (custom details) on `ApolloError`
 
-In Apollo Server 2, [error extensions](http://localhost:8000/data/errors/#including-custom-error-details) could be passed either as the third argument to the constructor, *or* as the `extensions` option on the third argument. That is, `new ApolloError(message, code, {key: 'value'})` was equivalent to `new ApolloError(message, code, {extensions: {key: 'value'}})`. No other "options" were recognized on the third argument. In Apollo Server 3, only the first form is supported. (If you try to use the second form, the constructor will throw an error.) Replace any code using the second form with the first form; you can do this before upgrading.
+#### Initializing an error
+
+In Apollo Server 2, [error extensions](http://localhost:8000/data/errors/#including-custom-error-details) can be passed to the `ApolloError` constructor either as the third argument, *or* as an `extensions` option on the third argument.
+
+In other words, these two lines are equivalent with Apollo Server 2:
 
-In Apollo Server 2, an error extension `foo` would show up on an error object `e` both as `e.foo` and as `e.extensions.foo`; when serialized for JSON error responses, it would show up both as `extensions.foo` and `extensions.exception.foo`. In Apollo Server 3, it only shows up in one place: on `e.extensions.foo` on the error object, and in `extensions.foo` when serialized. See [the PR](https://github.com/apollographql/apollo-server/pull/5294) for more details.
+```js
+new ApolloError(message, code, {key: 'value'}
+new ApolloError(message, code, {extensions: {key: 'value'}})
+```
+
+In Apollo Server 3, only the first line above is supported. If you try to use the second line, the constructor throws an error. Before upgrading, replace any code using the second form with the first form.
+
+#### Reading error extensions
+
+In Apollo Server 2, an error extension `foo` is present in two locations on an `ApolloError` object `error`: `error.foo` _and_ `error.extensions.foo`.
+
+In Apollo Server 3, it only shows up in one location: on `error.extensions.foo`. See [the PR](https://github.com/apollographql/apollo-server/pull/5294) for more details.
```

Comment:
You removed the part about where the extensions show up in serialized JSON responses, which was weirdly different from where they show up on the JS object (this odd `extensions.exception` thing).  (And most people who care about this stuff are concerned about the JSON response, ie what they see in their client)

Reply:
Ah yep missed the distinctions between the two, was clearly reading too fast